GitUI/UserEnvironmentInformation.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Windows.Forms;
+using GitCommands;
+
+namespace GitUI
+{
+    public sealed class UserEnvironmentInformation
+    {
+        public static void CopyInformation()
+        {
+            Clipboard.SetText(GetInformation());
+        }
+
+        public static string GetInformation()
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.AppendFormat("- Git Extensions version:  {0} ({1}){2}", AppSettings.ProductVersion, ThisAssembly.Git.Commit, Environment.NewLine);
```

Comment:
Where is `ThisAssembly.Git.Commit` defined?

Reply:
https://github.com/kzu/GitInfo

Reply:
```c#
// <auto-generated />
#define GLOBALNAMESPACE
#define NOMETADATA
#pragma warning disable 0436

#if ADDMETADATA
[assembly: System.Reflection.AssemblyMetadata("GitInfo.IsDirty", ThisAssembly.Git.IsDirtyString)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.Branch", ThisAssembly.Git.Branch)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.Commit", ThisAssembly.Git.Commit)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.Sha", ThisAssembly.Git.Sha)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.BaseVersion.Major", ThisAssembly.Git.BaseVersion.Major)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.BaseVersion.Minor", ThisAssembly.Git.BaseVersion.Minor)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.BaseVersion.Patch", ThisAssembly.Git.BaseVersion.Patch)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.Commits", ThisAssembly.Git.Commits)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.Tag", ThisAssembly.Git.Tag)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.BaseTag", ThisAssembly.Git.BaseTag)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.Major", ThisAssembly.Git.SemVer.Major)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.Minor", ThisAssembly.Git.SemVer.Minor)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.Patch", ThisAssembly.Git.SemVer.Patch)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.Label", ThisAssembly.Git.SemVer.Label)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.DashLabel", ThisAssembly.Git.SemVer.DashLabel)]
[assembly: System.Reflection.AssemblyMetadata("GitInfo.SemVer.Source", ThisAssembly.Git.SemVer.Source)]
#endif

#if LOCALNAMESPACE
namespace 
{
#endif
  /// <summary>Provides access to the current assembly information.</summary>
  partial class ThisAssembly
  {
    /// <summary>Provides access to the git information for the current assembly.</summary>
    public partial class Git
    {
      /// <summary>IsDirty: true</summary>
      public const bool IsDirty = true;

      /// <summary>IsDirtyString: true</summary>
      public const string IsDirtyString = "true";

      /// <summary>Branch: IssueEnvironment</summary>
      public const string Branch = "IssueEnvironment";

      /// <summary>Commit: ec28ebc05</summary>
      public const string Commit = "ec28ebc05";

      /// <summary>Sha: ec28ebc05f77381ed7556fd940aca336beafb649</summary>
      public const string Sha = "ec28ebc05f77381ed7556fd940aca336beafb649";

      /// <summary>Commits on top of base version: 286</summary>
      public const string Commits = "286";

      /// <summary>Tag: v3.00.00-beta1-286-gec28ebc05</summary>
      public const string Tag = "v3.00.00-beta1-286-gec28ebc05";

      /// <summary>Base tag: v3.00.00-beta1</summary>
      public const string BaseTag = "v3.00.00-beta1";

      /// <summary>Provides access to the base version information used to determine the <see cref="SemVer" />.</summary>      
      public partial class BaseVersion
      {
        /// <summary>Major: 3</summary>
        public const string Major = "3";

        /// <summary>Minor: 00</summary>
        public const string Minor = "00";

        /// <summary>Patch: 00</summary>
        public const string Patch = "00";
      }

      /// <summary>Provides access to SemVer information for the current assembly.</summary>
      public partial class SemVer
      {
        /// <summary>Major: 3</summary>
        public const string Major = "3";

        /// <summary>Minor: 00</summary>
        public const string Minor = "00";

        /// <summary>Patch: 286</summary>
        public const string Patch = "286";

        /// <summary>Label: beta1</summary>
        public const string Label = "beta1";

        /// <summary>Label with dash prefix: -beta1</summary>
        public const string DashLabel = "-beta1";

        /// <summary>Source: Tag</summary>
        public const string Source = "Tag";
      }
    }
  }
#if LOCALNAMESPACE
}
#endif
#pragma warning restore 0436
```

Reply:
I think we need to reflected that we are taking a new dependency to generate the current assembly info. At least in the commit message, but better in the code (if appropriate).